File: Components/WinForms/VB/SpreadsheetMainDemo/Modules/HeaderFooter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.IO
Imports DevExpress.XtraSpreadsheet.Demos
Imports DevExpress.Spreadsheet
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraSpreadsheet.Services
Imports DevExpress.XtraSpreadsheet.Commands

Namespace DevExpress.XtraSpreadsheet.Demos
	Partial Public Class HeaderFooterModule
		Inherits SpreadSheetTutorialControlBase
		Public Sub New()
			InitializeComponent()

			spreadsheetControl1.LoadDocument(DemoUtils.GetRelativePath("HeaderFooter_template.xlsx"), DocumentFormat.Xlsx)
			RefreshPreview()
		End Sub

		Private Sub btnPageSetup_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnPageSetup.Click
			Dim service As ISpreadsheetCommandFactoryService = CType(spreadsheetControl1.GetService(GetType(ISpreadsheetCommandFactoryService)), ISpreadsheetCommandFactoryService)
			Dim command As SpreadsheetCommand = service.CreateCommand(SpreadsheetCommandId.PageSetupHeaderFooter)
			command.ForceExecute(command.CreateDefaultCommandUIState())
			RefreshPreview()
		End Sub

		Private Sub RefreshPreview()
			Using ms As New MemoryStream()
				spreadsheetControl1.SaveDocument(ms, DocumentFormat.Xlsx)
				ms.Position = 0
				Dim workbook As New Workbook()
				workbook.LoadDocument(ms, DocumentFormat.Xlsx)
				Me.spreadsheetPreview1.Workbook = workbook
				Me.spreadsheetPreview1.UpdatePreview()
			End Using
		End Sub
	End Class
End Namespace

File: Components/WinForms/CS/SpreadsheetMainDemo/Modules/HeaderFooter.cs
using System;
using System.IO;
using DevExpress.XtraSpreadsheet.Demos;
using DevExpress.Spreadsheet;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraSpreadsheet.Services;
using DevExpress.XtraSpreadsheet.Commands;

namespace DevExpress.XtraSpreadsheet.Demos {
    public partial class HeaderFooterModule : SpreadSheetTutorialControlBase {
        public HeaderFooterModule() {
            InitializeComponent();

            spreadsheetControl1.LoadDocument(DemoUtils.GetRelativePath("HeaderFooter_template.xlsx"), DocumentFormat.Xlsx);
            RefreshPreview();
        }

        private void btnPageSetup_Click(object sender, EventArgs e) {
            ISpreadsheetCommandFactoryService service = (ISpreadsheetCommandFactoryService)spreadsheetControl1.GetService(typeof(ISpreadsheetCommandFactoryService));
            SpreadsheetCommand command = service.CreateCommand(SpreadsheetCommandId.PageSetupHeaderFooter);
            command.ForceExecute(command.CreateDefaultCommandUIState());
            RefreshPreview();
        }

        void RefreshPreview() {
            using (MemoryStream ms = new MemoryStream()) {
                spreadsheetControl1.SaveDocument(ms, DocumentFormat.Xlsx);
                ms.Position = 0;
                Workbook workbook = new Workbook();
                workbook.LoadDocument(ms, DocumentFormat.Xlsx);
                this.spreadsheetPreview1.Workbook = workbook;
                this.spreadsheetPreview1.UpdatePreview();
            }
        }
    }
}
